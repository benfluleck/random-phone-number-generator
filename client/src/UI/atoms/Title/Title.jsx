import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import setClassNames from '../../../helpers/utils/setClassNames';

const Title = ({ children }) => (
  <Title.Container className={setClassNames({ 'is-size-1': true, 'has-text-centered': true })}>
    {children}
  </Title.Container>
);

export default Title;

Title.Container = styled.h2``;

Title.propTypes = {
  children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
};
