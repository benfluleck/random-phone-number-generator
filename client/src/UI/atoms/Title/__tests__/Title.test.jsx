import React from 'react';

import { shallow } from 'enzyme';
import Title from '../Title';

const setup = (props) => {
  const wrapper = shallow(
    <Title {...props}>
      Test Title
    </Title>
  );

  return {
    wrapper,
  };
};


describe('Component - Button', () => {
  test('should render', () => {
    const { wrapper } = setup();

    expect(wrapper.exists()).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
