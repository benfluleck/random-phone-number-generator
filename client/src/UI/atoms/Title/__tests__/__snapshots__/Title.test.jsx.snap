// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - Button should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Title>
    Test Title
  </Title>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": "Test Title",
      "className": "is-size-1 has-text-centered",
    },
    "ref": null,
    "rendered": "Test Title",
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [],
      },
      "displayName": "styled.h2",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "h2",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": "Test Title",
        "className": "is-size-1 has-text-centered",
      },
      "ref": null,
      "rendered": "Test Title",
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [],
        },
        "displayName": "styled.h2",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "h2",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
